### PR TITLE
Mailet 149 raw ical + from

### PR DESCRIPTION
**Pull request by chibenwa**
https://github.com/linagora/james-project/pull/578

### CODE REVIEW

File: mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java [hunk 1]
```diff
@@ -197,4 +204,19 @@ public void service(Mail mail) throws MessagingException {
             return Stream.of();
         }
     }
+
+    private Optional<String> retrieveSender(Mail mail) throws MessagingException {
+        Optional<String> from = Optional.ofNullable(mail.getMessage())
+            .map(Throwing.function(MimeMessage::getFrom).orReturn(new Address[]{}))
+            .map(Arrays::stream)
+            .orElse(Stream.of())
+            .map(address -> (InternetAddress) address)
```
Comment (chibenwa) [inline mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java]: Can not avoid this cast as abstract class do not unbox addresses like : 

```
OBM <obm.org>
```

But InternetAddress can.

**Comment by chibenwa** [comment]
```
[Building Java 8 mailet-149-raw-ical-80786968-23ca-4741-a114-1c06e0fe3fd3] 
[Building Java 8 mailet-149-raw-ical-80786968-23ca-4741-a114-1c06e0fe3fd3] Tests in error: 
[Building Java 8 mailet-149-raw-ical-80786968-23ca-4741-a114-1c06e0fe3fd3]   Unauthorized
[Building Java 8 mailet-149-raw-ical-80786968-23ca-4741-a114-1c06e0fe3fd3]   Unauthorized
[Building Java 8 mailet-149-raw-ical-80786968-23ca-4741-a114-1c06e0fe3fd3] 
[Building Java 8 mailet-149-raw-ical-80786968-23ca-4741-a114-1c06e0fe3fd3] Tests run: 630, Failures: 0, Errors: 2, Skipped: 3
```

**Comment by chibenwa** [comment]
test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java [hunk 2]
```diff
@@ -56,9 +54,7 @@ public AmqpRule(SwarmGenericContainer rabbitMqContainer, String exchangeName, St
 
     @Override
     protected void before() throws Throwable {
-        @SuppressWarnings("deprecation")
-        InetAddress containerIp = InetAddresses.forString(rabbitMqContainer.getContainerInfo().getNetworkSettings().getIpAddress());
-        amqpUri = "amqp://" + containerIp.getHostAddress();
+       amqpUri = "amqp://" + rabbitMqContainer.getIp();
```
Comment (rouazana) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java]: indent?

File: server/mailet/integration-testing/src/test/resources/eml/yahooInvitation.eml [hunk 3]
```diff
@@ -1,193 +1,162 @@
...
- FURToxOTIzMDQyMVQyMzAwMDANClJEQVRFOjE5MjQwMzI5VDIzMDAwMA0KUkRBVEU6MTkyNTA0MDRUMjMwMDAwDQpSREFURToxOTI2MDQxN1QyMzAwMDANClJEQVRFOjE5MjcwNDA5VDIzMDAwMA0KUkRBVEU6MTkyODA0MTRUMjMwMDAwDQpSREFURToxOTI5MDQyMVQwMjAwMDANClJEQVRFOjE5MzAwNDEzVDAyMDAwMA0KUkRBVEU6MTkzMTA0MTlUMDIwMDAwDQpSREFURToxOTMyMDQwM1QwMjAwMDANClJEQVRFOjE5MzMwMzI2VDAyMDAwMA0KUkRBVEU6MTkzNDA0MDhUMDIwMDAwDQpSREFURToxOTM1MDMzMVQwMjAwMDANClJEQVRFOjE5MzYwNDE5VDAyMDAwMA0KUkRBVEU6MTkzNzA0MDRUMDIwMDAwDQpSREFURToxOTM4MDMyN1QwMjAwMDANClJEQVRFOjE5MzkwNDE2VDAyMDAwMA0KUkRBVEU6MTk0MDAyMjVUMDIwMDAwDQpFTkQ6REFZTElHSFQNCkJFR0lOOkRBWUxJR0hUDQpUWk9GRlNFVEZST006KzAyMDANClRaT0ZGU0VUVE86KzAyMDANClRaTkFNRTpDRVNUDQpEVFNUQVJUOjE5NDQwOTAzVDAwMDAwMA0KUkRBVEU6MTk0NDA5MDNUMDAwMDAwDQpFTkQ6REFZTElHSFQNCkJFR0lOOlNUQU5EQVJEDQpUWk9GRlNFVEZST006KzAxMDANClRaT0ZGU0VUVE86KzAxMDANClRaTkFNRTpDRVQNCkRUU1RBUlQ6MTk3NzAxMDFUMDAwMDAwDQpSREFURToxOTc3MDEwMVQwMDAwMDANCkVORDpTVEFOREFSRA0KRU5EOlZUSU1FWk9ORQ0KRU5EOlZDQUxFTkRBUg0K
---_004_CD15A92E9002734AADB6F2075789C4D70108100CB8EDEGLEX07VS01_--
+Return-Path: <do-not-reply@yahoo.com>
+Date: 23 Jan 2017 12:16:35 +0000
+To: ouazana@linagora.com
```
Comment (rouazana) [inline server/mailet/integration-testing/src/test/resources/eml/yahooInvitation.eml]: while you are anonymising, can you put a fake address? This one is also mine ^^

Comment (chibenwa) [inline server/mailet/integration-testing/src/test/resources/eml/yahooInvitation.eml]: Nope, I removed the "r" so that you don't receive SPAM

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**